code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,233 @@
+					icon_state_to_use = "full"
+					space_found = TRUE
+
+				if(!space_found)
+					switch(length(directions_to_aliens))
+						if(0)
+							icon_state_to_use = "full"
+						if(1)
+							dir_to_use = directions_to_aliens[1]
+						if(2)
+							if((directions_to_aliens[1] == NORTH && directions_to_aliens[2] == SOUTH) || (directions_to_aliens[1] == SOUTH && directions_to_aliens[2] == NORTH))
+								dir_to_use = NORTH
+							else if((directions_to_aliens[1] == EAST && directions_to_aliens[2] == WEST) || (directions_to_aliens[1] == WEST && directions_to_aliens[2] == EAST))
+								dir_to_use = EAST
+							else
+								if(directions_to_aliens[1] == NORTH)
+									if(directions_to_aliens[2] == EAST)
+										dir_to_use = NORTHEAST
+									else
+										dir_to_use = NORTHWEST
+								else if(directions_to_aliens[1] == SOUTH)
+									if(directions_to_aliens[2] == EAST)
+										dir_to_use = SOUTHEAST
+									else
+										dir_to_use = SOUTHWEST
+								else if(directions_to_aliens[1] == EAST )
+									if(directions_to_aliens[2] == NORTH)
+										dir_to_use = NORTHEAST
+									else
+										dir_to_use = SOUTHEAST
+								else
+									if(directions_to_aliens[2] == NORTH)
+										dir_to_use = NORTHWEST
+									else
+										dir_to_use = SOUTHWEST
+						if(3)
+							icon_state_to_use = "full"
+						if(4)
+							icon_state_to_use = "full"
+
+			else //if its space then nothing
+				continue
+
+			var/icon/I = icon(icon_to_use, icon_state_to_use, dir_to_use)
+
+			I.SwapColor(COLOR_WHITE, get_area_color(A))
+			I.SwapColor(COLOR_BLACK, get_area_color(A, TRUE))
+
+			bigassicon.Blend(I, ICON_OVERLAY, x_index*2, y_index*2)
+
+	return bigassicon
+
+
+
+
+
+/obj/machinery/holomap/proc/get_area_color(area/A, floor_varient=FALSE)
+	//don't look too closely at this, I blame the mappers
+
+	//SHUTTLES
+	if(istype(A, /area/shuttle))
+		return floor_varient ? "#bdbdbd" : "#FFFFFF"
+
+	//AI SAT
+	if(istype(A, /area/turret_protected) || istype(A, /area/aisat))
+		return floor_varient ? "#144640": "#009180"
+
+	//COMMAND
+	if(istype(A, /area/bridge) || (istype(A, /area/teleporter) && !istype(A, /area/teleporter/quantum)) || istype(A, /area/crew_quarters/heads) || istype(A, /area/crew_quarters/chief) || istype(A, /area/crew_quarters/captain) || istype(A, /area/ntrep) || istype(A, /area/blueshield) || istype(A, /area/ai_monitored) || istype(A, /area/security/nuke_storage) || istype(A, /area/server) || istype(A, /area/crew_quarters/hor) || istype(A, /area/solar)/*not a mistake, hijacking the color*/ || istype(A, /area/engine/chiefs_office) || istype(A, /area/medical/cmo) || istype(A, /area/security/hos))
+		return floor_varient ? "#1e1a5e": "#08009f"
+
+	//SECURITY
+	if((istype(A, /area/security) && !(istype(A, /area/security/vacantoffice) || istype(A, /area/security/vacantoffice2)))|| istype(A, /area/lawoffice) || istype(A, /area/crew_quarters/courtroom) || istype(A, /area/magistrateoffice))
```

Comment:
These are pretty rough checks, I'm pretty sure we have helpers for this (type in list, hell, possibly even a typecache) so you don't need to `istype` all these.

Reply:
I'm not just doing `if(istypes)` but also if(!istypes) since areas are just all a mess, so not so shrimple. If you could suggest something I could copy from or look at to understand what you mean maybe I could attempt something..